doc/tutorials/pick_and_place_with_moveit_task_constructor/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-add_executable(mtc_tutorial src/main.cpp)
```

Comment:
I have a PR (#455) under review that instructs the user to create a mtc_node.cpp file for the mtc_tutorial project, so I wanted the docker image to have the same file name. I thought it would make sense to rename main.cpp to match the tutorial naming, but I can perform the renaming operation as the image is being created instead of changing the original filename. I figured if the user plans to download the cpp file rather than writing or copying it I could save them the confusion of renaming it

Reply:
I expect people to have the tutorial sources downloaded anyway. Wouldn't the same filename induce a conflict then?

Reply:
My idea was to build an image following the steps in the tutorial as closely as possible. This way, building the image itself is a test of the tutorial instructions. I could more closely follow the tutorial instructions when building the image if I performed an operation that mimics the user pasting copied code in to their `mtc_node.cpp`, however I figured duplicating the `mtc_node.cpp` file from `moveit2_tutorials` was close enough and less likely to break if someone modifies the `moveit2_tutorials` file. I also wanted the image to be built so that if someone wanted to skip any or all of the tutorials and related setup they could pull the appropriate image and head straight to the final actions of the tutorial. I thought it made sense for consistency to have both files use the same name so I considered 3 options:

1. Change the tutorial to `ros2 pkg create --build-type ament_cmake --node-name main mtc_tutorial`
2. Rename `moveit2_tutorials/main.cpp` to `moveit2_tutorials/mtc_node.cpp`
3. Leave the 2 files with different names and change it in the Dockerfile

I think it is best to encourage the user to make changes to the tutorial's source files and it makes sense to have them do that in their own package rather than the `moveit2_tutorials` package so I believe duplicating `mtc_node.cpp` is a necessity, unless something like the following changes would be beneficial:

`COPY ./doc/tutorials/pick_and_place_with_moveit_task_constructor/src/mtc_node.cpp src/mtc_tutorial/src/mtc_node.cpp` 

`RUN cp src/moveit2_tutorials/doc/tutorials/pick_and_place_with_moveit_task_constructor/src/mtc_node.cpp src/mtc_tutorial/src/mtc_node.cpp`

Reply:
> I think it is best to encourage the user to make changes to the tutorial's source files and it makes sense to have them do that in their own package rather than the `moveit2_tutorials` package.

I don't agree with this statement, but this is a subjective point of view :wink:
Having understood your reasoning, I don't really care too much about which option you choose. If you rename the source file, ensure that all links to it remain valid!